packages/traceability/issuer-api/src/pods/certificate/handlers/batch-transfer-certificates.handler.ts

```diff
@@ -0,0 +1,68 @@
+import { CommandHandler, ICommandHandler } from '@nestjs/cqrs';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { CertificateBatchOperations } from '@energyweb/issuer';
+import { ISuccessResponse, ResponseFailure, ResponseSuccess } from '@energyweb/origin-backend-core';
+import { BigNumber } from 'ethers';
+import { HttpStatus } from '@nestjs/common';
+import { BatchTransferCertificatesCommand } from '../commands/batch-transfer-certificates.command';
+import { Certificate } from '../certificate.entity';
+import { BlockchainPropertiesService } from '../../blockchain/blockchain-properties.service';
+
+@CommandHandler(BatchTransferCertificatesCommand)
+export class BatchTransferCertificatesHandler
+    implements ICommandHandler<BatchTransferCertificatesCommand>
+{
+    constructor(
+        @InjectRepository(Certificate)
+        private readonly repository: Repository<Certificate>,
+        private readonly blockchainPropertiesService: BlockchainPropertiesService
+    ) {}
+
+    async execute({
+        certificateIds,
+        to,
+        values,
+        forAddress
+    }: BatchTransferCertificatesCommand): Promise<ISuccessResponse> {
+        const blockchainProperties = await this.blockchainPropertiesService.get();
+
+        const certificatesToTransfer = await this.repository.findByIds(certificateIds, {
+            relations: ['blockchain']
+        });
+
+        if (
+            !certificatesToTransfer.every(
+                (cert) => cert.owners[forAddress] && BigNumber.from(cert.owners[forAddress]).gt(0)
+            )
+        ) {
+            return ResponseSuccess('You have requested claiming of a certificate you do not own');
```

Comment:
I guess you can easily return the list of certificates ids that you don't own to simplify debugging (sending 10 transfers, and one of them is wrong, requires me to debug 10 times more)

BTW this is transfer, not claim